diff --git a/drivers/can/can_mcan.c b/drivers/can/can_mcan.c
index 3c7bdc233b..f00fe41b06 100644
--- a/drivers/can/can_mcan.c
+++ b/drivers/can/can_mcan.c
@@ -646,15 +646,15 @@ static void can_mcan_get_message(const struct device *dev, uint16_t fifo_offset,
 			goto ack;
 		}
 
-		if (((frame.flags & CAN_FRAME_FDF) != 0U && (flags & CAN_FILTER_FDF) == 0U) ||
-		    ((frame.flags & CAN_FRAME_FDF) == 0U && (flags & CAN_FILTER_FDF) != 0U)) {
-			/* FDF bit does not match filter, drop frame */
-			err = can_mcan_write_reg(dev, fifo_ack_reg, get_idx);
-			if (err != 0) {
-				return;
-			}
-			goto ack;
-		}
+		// if (((frame.flags & CAN_FRAME_FDF) != 0U && (flags & CAN_FILTER_FDF) == 0U) ||
+		//     ((frame.flags & CAN_FRAME_FDF) == 0U && (flags & CAN_FILTER_FDF) != 0U)) {
+		// 	/* FDF bit does not match filter, drop frame */
+		// 	err = can_mcan_write_reg(dev, fifo_ack_reg, get_idx);
+		// 	if (err != 0) {
+		// 		return;
+		// 	}
+		// 	goto ack;
+		// }
 
 		data_length = can_dlc_to_bytes(frame.dlc);
 		if (data_length <= sizeof(frame.data)) {
